### PR TITLE
feature/MA/helper-methods

### PR DESCRIPTION
**Pull request by Mohammad-Alavi**
Add helper methods to verify model relations. Greatly simplifies writing policies.

### CODE REVIEW

File: Traits/CanOwnTrait.php [hunk 1]
```diff
@@ -5,67 +5,64 @@
...
-     */
-    private function guessMorphs(Model $ownable): array
+    protected function guessSingularRelationshipName(Model $ownable): string
     {
-        $className = Str::snake(class_basename($ownable));
+        return Str::camel(class_basename($ownable));
+    }
 
-        return [$className . 'able_id', $className . 'able_type'];
+    protected function guessPlurarRelationshipName(Model $ownable): string
```
Comment (yalsicor) [inline Traits/CanOwnTrait.php]: typo

File: Traits/CanOwnTrait.php [hunk 2]
```diff
@@ -5,67 +5,64 @@
...
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Support\Str;
-use Throwable;
 
 trait CanOwnTrait
 {
     /**
-     * Checks if the model is the owner of the $ownable model
-     * by comparing IDs
+     * Checks if the model is the owned by the $owner model.
```
Comment (yalsicor) [inline Traits/CanOwnTrait.php]: minus 'the'

File: Traits/CanOwnTrait.php (hunk 1 above)
Comment (Mohammad-Alavi) [reply Traits/CanOwnTrait.php]: I cannot see the typo here. 👁️

File: Traits/CanOwnTrait.php (hunk 2 above)
Comment (Mohammad-Alavi) [reply Traits/CanOwnTrait.php]: Like this?
`Checks if model is the owned by the $owner model.`

File: Traits/CanOwnTrait.php (hunk 1 above)
Comment (yalsicor) [reply Traits/CanOwnTrait.php]: Plurar?

File: Traits/CanOwnTrait.php (hunk 2 above)
Comment (yalsicor) [reply Traits/CanOwnTrait.php]: Checks if the model is owned by the $owner model.

**Review by yalsicor — APPROVED** [review]

**Pull request merged**